Classes/Hooks/HookProvider.php

```diff
@@ -117,9 +117,11 @@ public function executeHookReturningMergedArray(string $method, ...$params): arr
 
         $result = [];
         foreach ($this->getHooks() as $hook) {
-            $result = array_merge($result, $hook->$method(...$params));
+            $result[] = $hook->$method(...$params);
         }
 
+        \array_merge(...$result);
```

Comment:
Shouldn't this be `$result = \array_merge(...$result);` or `return \array_merge(...$result);` (as the result of `array_merge` is passed as the return value)?